tests/loader_test_vectors.rs

```diff
@@ -61,20 +67,64 @@ fn concat<'a>(left: &[&'a str], right: &[&'a str]) -> Vec<&'a str> {
         .collect()
 }
 
+/// Determines if the given file name is in the paths list.  This deals with platform
+/// path separator differences from '/' separators in the path list.
+#[inline]
+fn contains_path(paths: &[&str], file_name: &str) -> bool {
+    paths
+        .into_iter()
+        // TODO construct the paths in a not so hacky way
+        .map(|p| p.replace("/", &PATH_SEPARATOR.to_string()))
+        .find(|p| p == file_name)
+        .is_some()
+}
+
 fn load_file<L: Loader>(loader: &L, file_name: &str) -> IonResult<Vec<OwnedElement>> {
     // TODO have a better API that doesn't require buffering into memory everything...
     let data = read(file_name)?;
-    loader.load_all(&data)
+    let result = loader.load_all(&data);
+
+    // do some simple single value loading tests
+    let single_result = loader.load_one(&data);
+    match &result {
+        Ok(elems) => {
+            if elems.len() == 1 {
+                match single_result {
+                    Ok(elem) => {
+                        // only compare if we know equality to work
+                        if !contains_path(LOAD_ONE_EQUIVS_SKIP_LIST, file_name) {
+                            assert_eq!(elems[0], elem)
+                        }
+                    }
+                    Err(e) => panic!("Expected element {:?}, got {:?}", elems, e),
+                }
+            } else {
+                match single_result {
+                    Ok(elem) => panic!(
+                        "Did not expect element for duplicates: {:?}, {:?}",
+                        elems, elem
+                    ),
+                    Err(e) => match e {
+                        IonError::DecodingError { description: _ } => (),
+                        other => {
+                            panic!("Got an error we did not expect for duplicates: {:?}", other)
+                        }
+                    },
+                }
+            }
+        }
+        Err(_) => assert!(
+            single_result.is_err(),
+            "Expected error from load_one: {:?}",
+            single_result
+        ),
```

Comment:
Ideally, we do stronger assertions here--but it is a bit complicated to make sure the errors line up consistently.